js/main.js

```diff
@@ -210,19 +213,22 @@ function handleHeart(target) {
       break;
     }
   }
-  for (i = 0; i < userDisplay.children.length; i++) {
-    for (j = 0; j < userDisplay.children[i].children.length; j++) {
-      if (userDisplay.children[i].children[j].getAttribute('id') === currentThumbID) {
-        userDisplay.children[i].children[j].lastChild.lastChild.firstChild.setAttribute('src', 'images/Empty_Heart.svg');
-      }
-    }
-  }
+  updateHearIcon('images/Empty_Heart.svg');
   if (destination.children.length === 0) {
     destination.append(nothingSaved(destination));
   }
 }
 
 function handleBookmark(target) {
+  function updateBookIcon(newImgURL) {
+    for (var i = 0; i < userDisplay.children.length; i++) {
+      for (var j = 0; j < userDisplay.children[i].children.length; j++) {
+        if (userDisplay.children[i].children[j].getAttribute('id') === currentThumbID) {
+          userDisplay.children[i].children[j].lastChild.firstChild.firstChild.setAttribute('src', newImgURL);
+        }
+      }
+    }
+  }
```

Comment:
Same thing here, if it's possible to move this function definition out of the `handleBookmark` function, that would be ideal.